### PR TITLE
[rtloader] Add doxygen config & rtloader.generate-doc invoke task

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Add basic doxygen config to `rtloader` to generate documentation,
and an invoke task which generates the docs and shows the warnings &
errors.

The invoke task could be used in the CI (it is setup to fail if
there are doxygen errors), but will need a runner with doxygen installed.

The CSS file is completely optional, I put it to prettify the HTML pages.

### Motivation

Being able to generate the documentation & verify that our code is correctly documented is good.

### Additional Notes

Open questions:
- Are there specific configuration options people would like enabled?
  - For instance, do we want to disable preprocessing?

### CODE REVIEW

File: rtloader/doxygen/CustomDoxygen.css [hunk 1]
```diff
@@ -0,0 +1,619 @@
+
+/* function descriptions */
```
Comment (ogaca-dd) [inline rtloader/doxygen/CustomDoxygen.css]: Any specific reason to use a custom css?

File: tasks/rtloader.py [hunk 2]
```diff
@@ -79,3 +79,50 @@ def format(ctx, raise_if_changed=False):
...
+    """
+    Generates the doxygen documentation, puts it in rtloader/doc, and logs doc errors/warnings.
+    (rtloader/doc is hardcoded right now in the Doxyfile, as doxygen cannot take the output directory as argument)
+    Logs all errors and warnings to <rtloader_path>/doxygen/errors.log and to the standard output.
+    Returns 1 if errors were found (by default, doxygen returns 0 even if errors are present).
+    """
+    rtloader_path = get_rtloader_path()
+
+    # doxygen puts both errors and warnings in stderr
+    result = ctx.run("doxygen {0}/doxygen/Doxyfile 2>{0}/doxygen/errors.log".format(rtloader_path), warn=True)
```
Comment (ogaca-dd) [inline tasks/rtloader.py]: If `rtloader_path` contains a space this command does not work.

File: tasks/rtloader.py [hunk 3]
```diff
@@ -79,3 +79,50 @@ def format(ctx, raise_if_changed=False):
...
+
+    if result.exited != 0:
+        print("Fatal error encountered while trying to generate documentation.")
+        print("See {0}/doxygen/errors.log for details.".format(rtloader_path))
```
Comment (ogaca-dd) [inline tasks/rtloader.py]: The error looks like: `See C:\XXX\datadog-agent\tasks\..\rtloader/doxygen/errors.log for details.`. You might normalize `rtloader_path`.

File: tasks/rtloader.py [hunk 4]
```diff
@@ -79,3 +79,50 @@ def format(ctx, raise_if_changed=False):
...
+        raise Exit(code=result.exited)
+
+    errors, warnings = [], []
+
+    def flushentry(entry):
```
Comment (ogaca-dd) [inline tasks/rtloader.py]: `entry` unused.

**Review by ogaca-dd — COMMENTED** [review]
Looks good. Try to handle spaces in the path is possible.

File: rtloader/doxygen/CustomDoxygen.css (hunk 1 above)
Comment (KSerrania) [reply rtloader/doxygen/CustomDoxygen.css]: No particular reason except that I prefer this theme to the base one. I can remove it (and the corresponding configuration option in the Doxyfile) if you think we should remove it

Comment (ogaca-dd) [reply rtloader/doxygen/CustomDoxygen.css]: It was just a question.

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**